packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -261,18 +262,78 @@ export class JobService {
     if ((manifest as CvatManifestDto)?.annotation?.type) {
       await this.sendWebhook(
         this.configService.get<string>(
-          ConfigNames.EXCHANGE_ORACLE_WEBHOOK_URL,
+          ConfigNames.CVAT_EXCHANGE_ORACLE_WEBHOOK_URL,
         )!,
         {
           escrowAddress: jobEntity.escrowAddress,
           chainId: jobEntity.chainId,
+          eventType: EventType.ESCROW_CREATED
         },
       );
     }
 
     return jobEntity;
   }
 
+  public async requestToCancelJob(
+    userId: number,
+    id: number,
+  ): Promise<boolean> {
+    const jobEntity = await this.jobRepository.findOne({ id, userId });
+    
+    if (!jobEntity) {
+      this.logger.log(ErrorJob.NotFound, JobService.name);
+      throw new NotFoundException(ErrorJob.NotFound);
+    }
+  
+    jobEntity.status = JobStatus.TO_CANCEL;
+    jobEntity.retriesCount = 0;
+    await jobEntity.save();
+    
+    return true;
+  }
+
+  public async cancelJob(
+    jobEntity: JobEntity
+  ): Promise<boolean> {
+    if (jobEntity.escrowAddress) {
```

Comment:
If the escrow has been created yet we need to return the payment to the job requester